Unable to inject S3Crt async client

When using the qualifier `@S3Crt` on the async client I'm getting

```
[ERROR]   S3ResourcesTest.testResource(String)[1] » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type software.amazon.awssdk.services.s3.S3AsyncClient and qualifiers [@S3Crt]
        - injection target: org.acme.s3.S3AsyncClientResource#s3
        - declared on CLASS bean [types=[org.acme.s3.S3AsyncClientResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.s3.S3AsyncClientResource]
        The following beans match by type, but none has matching qualifiers:
        - PRODUCER METHOD bean [types=[java.lang.AutoCloseable, software.amazon.awssdk.services.s3.S3AsyncClient, software.amazon.awssdk.core.SdkClient, java.lang.Object, software.amazon.awssdk.utils.SdkAutoCloseable, software.amazon.awssdk.awscore.AwsClient], qualifiers=[@Default, @Any], target=software.amazon.awssdk.services.s3.S3AsyncClient asyncClient(), declaringBean=io.quarkus.amazon.s3.runtime.S3ClientProducer]
```

Steps to reproduce:
Download the quarkus quickstart and in the `amazon-s3-quickstart` project follow the async s3crt directions:

Add

```
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>aws-crt-client</artifactId>
        </dependency>
```

Configure

```
quarkus.s3.async-client.type=aws-crt
```

In class `S3AsyncClientResource` add the qualifer `S3Crt` to

```
    @Inject
    @S3Crt
    S3AsyncClient s3;

```

Run tests


Versions:

Quarkus 3.12.3
Quarkus Amazon 2.16.0
maven  3.9.6 
Java 21.0.4

Follow up: Issue also happens with sync client.

--- a/s3/deployment/src/main/java/io/quarkus/amazon/s3/deployment/S3CrtProcessor.java
+++ b/s3/deployment/src/main/java/io/quarkus/amazon/s3/deployment/S3CrtProcessor.java
@@ -69,7 +69,9 @@ public class S3CrtProcessor {
 
             Type injectedType = getInjectedType(injectionPoint);
 
-            if (asyncClientName().equals(injectedType.name())) {
+            // Check if the injected type is S3AsyncClient
+            DotName s3AsyncClientName = DotName.createSimple(S3AsyncClient.class.getName());
+            if (s3AsyncClientName.equals(injectedType.name())) {
                 asyncClassName = Optional.of(asyncClientName());
             }
         }

--- a/s3/deployment/src/main/java/io/quarkus/amazon/s3/deployment/S3CrtProcessor.java
+++ b/s3/deployment/src/main/java/io/quarkus/amazon/s3/deployment/S3CrtProcessor.java
@@ -36,6 +36,7 @@ import software.amazon.awssdk.services.s3.internal.crt.S3CrtAsyncClient;
 public class S3CrtProcessor {
 
     public static final DotName S3CRT = DotName.createSimple(S3Crt.class);
+    private static final DotName S3_ASYNC_CLIENT = DotName.createSimple(S3AsyncClient.class.getName());
 
     S3BuildTimeConfig buildTimeConfig;
 
@@ -69,9 +70,7 @@ public class S3CrtProcessor {
 
             Type injectedType = getInjectedType(injectionPoint);
 
-            // Check if the injected type is S3AsyncClient
-            DotName s3AsyncClientName = DotName.createSimple(S3AsyncClient.class.getName());
-            if (s3AsyncClientName.equals(injectedType.name())) {
+            if (S3_ASYNC_CLIENT.equals(injectedType.name())) {
                 asyncClassName = Optional.of(asyncClientName());
             }
         }

